File: 0049-group-anagrams/0049-group-anagrams.go
func groupAnagrams(strs []string) [][]string {

	groups := make(map[string][]string)

	for i := range (len(strs)) {
		intBuffer := make([]int, len(strs[i]))

		for j := range strs[i] {
			intBuffer[j] = int(strs[i][j])
		}
		slices.Sort(intBuffer)

		byteBuffer := make([]byte, len(strs[i]))

		for k := range len(intBuffer) {
			byteBuffer[k] = byte(intBuffer[k])
		}

		groups[string(byteBuffer)] = append(groups[string(byteBuffer)], strs[i])
	}

	result := [][]string{}

	for _, v := range groups {
		result = append(result, v)
	}

	return result
}

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {

        unordered_map<string, vector<string>> sortedMap;

        for (const string &s : strs) {
            string sortedS = s;
            sort(sortedS.begin(), sortedS.end());
            sortedMap[sortedS].push_back(s);
        }

        vector<vector<string>> result;

        for (const pair<string, vector<string>> &entry : sortedMap) {
            result.push_back(entry.second);
        }

        return result;
    }
};